Nodejs/Product/TestAdapter/TestFrameworks/Angular/Angular.js

```diff
@@ -156,18 +157,18 @@ const run_tests = async function (context) {
 }
 
 function detectPackage(projectFolder, packageName) {
-    try {
-        const packagePath = path.join(projectFolder, 'node_modules', packageName);
-        const pkg = require(packagePath);
+    const packagePath = path.join(projectFolder, 'node_modules', packageName);
 
-        return pkg;
-    } catch (ex) {
+    if (!fs.existsSync(packagePath)) {
```

Comment:
`existsSync` won't do the same module resolution as `require` or handle things like matching `index.js` if the argument is a directory.  Does that matter?

Reply:
It matters, `existsSync` wont load the file, it will only check it it exists. `require` will load it memory, execute and return the export. The latter will unnecessarily consume more resources, so I refactored.

Reply:
I think `require` does two things that `existsSync` does not: it loads the file - which, I agree, is undesirable - but it also does extra resolution to find the file and I don't know whether that's important.  If we're passing an absolute path to a file (not a directory), it probably isn't.

Reply:
I suppose if we want to be super thorough on checking the dependencies that behavior might be desirable.  We are looking into the directory. I admit I don't have enough knowledge to now how npm resolve things, I'm not confident to change it until looking further.

IMO, I don't think we need to be that meticulous when checking for dependencies as this is only for showing a custom message if they don't exists. If for any reason the directory exists but not the package, we still get an error message anyway. 

Let me know if you think otherwise, I can change it back if you feel it adds some value to have.